test/e2e/e2e_test.go

```diff
@@ -0,0 +1,286 @@
+				"event.message":                   "Pulling image \"" + testpod.Spec.Containers[0].Image + "\"",
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+		{
+			Summary:  "Successfully pulled image \"" + testpod.Spec.Containers[0].Image + "\"",
+			Category: "kubernetes",
+			Attributes: map[string]interface{}{
+				"event.metadata.name":             testpod.Name + ".",
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Pulled",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Successfully pulled image \"" + testpod.Spec.Containers[0].Image + "\"",
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+		{
+			Summary:  "Created container " + testpod.Spec.Containers[0].Name,
+			Category: "kubernetes",
+			Attributes: map[string]interface{}{
+				"event.metadata.name":             testpod.Name + ".",
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Created",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Created container " + testpod.Spec.Containers[0].Name,
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+		{
+			Summary:  "Started container " + testpod.Spec.Containers[0].Name,
+			Category: "kubernetes",
+			Attributes: map[string]interface{}{
+				"event.metadata.name":             testpod.Name + ".",
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Started",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Started container " + testpod.Spec.Containers[0].Name,
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+	} {
+		if agentMock.Has(&event) {
+			continue
+		}
+
+		e := json.NewEncoder(os.Stderr)
+		t.Log("Expected:")
+		_ = e.Encode(event)
+		t.Log("Have:")
+		_ = e.Encode(agentMock.Events())
+		t.Fatalf("Event was not captured")
+	}
+}
+
+func TestPodDeletion(t *testing.T) {
+	client, agentMock := initialize(t)
+
+	t.Log("Creating test namespace...")
+	ns, err := client.CoreV1().Namespaces().Create(context.Background(), &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "e2e-" + strings.ToLower(t.Name()) + "-",
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		t.Logf("could not create %s namespace: %v", ns, err)
+	}
```

Comment:
I suggest to de-duplicate this to `withTestNamespace(t *testing.T) string` function. This make test more readable.